src/redteam/commands/generate.ts

```diff
@@ -142,62 +130,40 @@ export async function doGenerateRedteam(options: RedteamCliGenerateOptions) {
     plugins: plugins || [],
   };
 
-  if (options.output) {
-    const existingYaml = configPath
-      ? (yaml.load(fs.readFileSync(configPath, 'utf8')) as Partial<UnifiedConfig>)
-      : {};
-    const updatedYaml: Partial<UnifiedConfig> = {
-      ...existingYaml,
-      defaultTest: {
-        ...(existingYaml.defaultTest || {}),
-        metadata: {
-          ...(existingYaml.defaultTest?.metadata || {}),
-          purpose,
-          entities,
-        },
-      },
-      tests: redteamTests,
-      redteam: { ...(existingYaml.redteam || {}), ...updatedRedteamConfig },
-    };
-    writePromptfooConfig(updatedYaml, options.output);
-    printBorder();
-    const relativeOutputPath = path.relative(process.cwd(), options.output);
-    logger.info(`Wrote ${redteamTests.length} new test cases to ${relativeOutputPath}`);
-    logger.info(
-      '\n' +
-        chalk.green(
-          `Run ${chalk.bold(
-            relativeOutputPath === 'redteam.yaml'
-              ? 'promptfoo redteam eval'
-              : `promptfoo redteam eval -c ${relativeOutputPath}`,
-          )} to run the red team!`,
-        ),
-    );
-    printBorder();
-  } else if (options.write && configPath) {
-    const existingConfig = yaml.load(fs.readFileSync(configPath, 'utf8')) as Partial<UnifiedConfig>;
-    existingConfig.defaultTest = {
-      ...(existingConfig.defaultTest || {}),
+  const command =
+    path.relative(process.cwd(), configPath) === 'promptfooconfig.yaml'
+      ? 'promptfoo redteam eval'
+      : `promptfoo redteam eval -c ${path.relative(process.cwd(), configPath)}`;
+
+  const existingYaml = configPath
+    ? (yaml.load(fs.readFileSync(configPath, 'utf8')) as Partial<UnifiedConfig>)
+    : {};
+  const updatedYaml: Partial<UnifiedConfig> = {
+    ...existingYaml,
+    defaultTest: {
+      ...(existingYaml.defaultTest || {}),
       metadata: {
-        ...(existingConfig.defaultTest?.metadata || {}),
+        ...(existingYaml.defaultTest?.metadata || {}),
         purpose,
         entities,
       },
-    };
-    existingConfig.tests = [...(existingConfig.tests || []), ...redteamTests];
-    existingConfig.redteam = { ...(existingConfig.redteam || {}), ...updatedRedteamConfig };
-    writePromptfooConfig(existingConfig, configPath);
-    logger.info(
-      `\nWrote ${redteamTests.length} new test cases to ${path.relative(process.cwd(), configPath)}`,
-    );
-    const command = configPath.endsWith('promptfooconfig.yaml')
-      ? 'promptfoo eval'
-      : `promptfoo eval -c ${path.relative(process.cwd(), configPath)}`;
-    logger.info('\n' + chalk.green(`Run ${chalk.bold(`${command}`)} to run the red team!`));
-  } else {
-    writePromptfooConfig({ tests: redteamTests }, 'redteam.yaml');
-  }
+    },
+    tests: [
+      ...new Set([
+        ...(existingYaml?.tests || []),
```

Comment:
I would like to preserve the rest of the promptfooconfig file exactly, so let's try regexing this :scream: 

We need to handle the following cases:
- `tests: ...` is a oneliner
- `tests:` exists and has a bunch of indented stuff under it
- `tests` key does not exist